src/Courses/TopicGrades/GradingPage.tsx

```diff
@@ -0,0 +1,259 @@
+        // when user and problem are selected - set the available workbooks and 
+        // pick one workbook as the default for rendering
+        // TODO: adjust for different policies -- best individual / best attempt
+        let currentPath: string | undefined;
+        let currentSeed: number | undefined;
+        let currentUserGrade: StudentGradeDict | undefined;
+        let currentWorkbooks: Record<number, StudentWorkbookInterface> | undefined;
+        let currentWorkbook: StudentWorkbookInterface | undefined;
+        
+        if (!_.isNil(selected.problem) && !_.isNil(selected.user)) {
+            if (!_.isNil(selected.problem.webworkQuestionPath)) {
+                currentPath = selected.problem.webworkQuestionPath;
+            } 
+            if (!_.isNil(selected.problem.id) && !_.isNil(selected.user.id)) {
+                const problemDict = problemMap[selected.problem.id];
+                if (!_.isNil(problemDict.grades)) {
+                    currentUserGrade = problemDict.grades[selected.user.id];
+                    if (!_.isNil(currentUserGrade.randomSeed)) {
+                        currentSeed = currentUserGrade.randomSeed;
+                    }
+                    const userProblemWorkbooks = currentUserGrade.workbooks;
+                    const selectedWorkbookId = currentUserGrade.lastInfluencingAttemptId; // this can be expanded
+                    if (!_.isNil(userProblemWorkbooks) && (!_.isEmpty(userProblemWorkbooks))) {
+                        currentWorkbooks = userProblemWorkbooks;
+                        if (!_.isNil(selectedWorkbookId)) {
+                            currentWorkbook = userProblemWorkbooks[selectedWorkbookId];
+                            if (_.isNil(currentWorkbook)) {
+                                logger.error(`we were supposed to get workbook #${selectedWorkbookId}, but failed.`);
+                            } else {
+                                // logger.debug(currentWorkbook.id);
+                                currentPath = undefined;
+                                currentSeed = undefined;
+                            }
+                        } else {
+                            logger.error(`student #${selected.user.id} has workbooks for problem #${selected.problem.id} but no target-able id`);
+                            logger.debug(currentUserGrade);
+                        }
+                    } else {
+                        // no error - student simply has no workbooks
+                        // what do we set to render instead?
+                        logger.info('student has no workbooks for this problem');
+                    }
+                } else {
+                    logger.error('will anything ever trigger this? It is a problem with NO grades?!');
+                }
+            } else {
+                logger.error('User and problem are selected, but one is missing an id!');
```

Comment:
In the case of these errors, we're still setting the selected info?

Reply:
error state handling has now been added